src/rational_bezier_spline.inc

```diff
@@ -134,39 +134,43 @@ constexpr PhysicalPointType RationalBezierSpline<
 
 template <std::size_t parametric_dimension, typename PhysicalPointType,
           typename ScalarType>
-constexpr std::array<std::vector<ScalarType>, parametric_dimension>
-RationalBezierSpline<parametric_dimension, PhysicalPointType, ScalarType>::
-    BasisFunctions(const PointTypeParametric_& par_coords) const {
+constexpr std::vector<ScalarType> RationalBezierSpline<
+    parametric_dimension, PhysicalPointType,
+    ScalarType>::BasisFunctions(const PointTypeParametric_& par_coords) const {
   // Check sanity
   assert(CheckSplineCompatibility());
+
+  // Basis function of weight function and weighted spline are equivalent
   // Evaluiere Basis Functions
-  std::array<std::vector<ScalarType>, parametric_dimension> factors =
-      weighted_spline_.BasisFunctions(par_coords);
+  std::vector<ScalarType> factors = weight_function_.BasisFunctions(par_coords);
 
   // Initialize values
   ScalarType weight_eval{};
-  weight_function_.AddUpContributionsToControlPointVector_(
-      weight_eval, factors, static_cast<ScalarType>(1.));
+  for (IndexingType i_basis_function{0}; i_basis_function < factors.size();
+       i_basis_function++) {
+    weight_eval += factors[i_basis_function] * GetWeights()[i_basis_function];
```

Comment:
```suggestion
    weight_eval += factors[i_basis_function] * GetWeights()[i_basis_function];
    factors[i_basis_function] *= GetWeights()[i_basis_function];
```
?

Reply:
That really depends on the application... As we multiply with the weighted control points later, this might become a problem. Maybe we need to provide another function for unweighted rational basis functions? What do you think?

Reply:
this won't pass partition of unity

Reply:
if we return basis functions, they should have partition of unity properties. For internal use, I guess it is your choice, but I think the function should return what it says. Just like rational beziers have `GetWeightedControlPoints()`, not `GetControlPoints()`